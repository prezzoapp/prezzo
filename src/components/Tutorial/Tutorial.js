// @flow
import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import Swiper from 'react-native-swiper';
import TutorialScreen from './TutorialScreen';
import Button from '../Button';
import FacebookButton from '../FacebookButton';
import {FONT_FAMILY} from '../../services/constants';

const images = {
  tutorial1: require('../../../assets/images/tutorial/tutorial-1.png'),
  tutorial2: require('../../../assets/images/tutorial/tutorial-2.png'),
  tutorial3: require('../../../assets/images/tutorial/tutorial-3.png'),
  tutorial4: require('../../../assets/images/tutorial/tutorial-4.png')
};

class Tutorial extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Swiper
          loop={false}
          dotColor='rgba(255, 255, 255, 0.5)'
          activeDotColor='#0DD24A'
        >
          <TutorialScreen image={images.tutorial1} />
          <TutorialScreen image={images.tutorial2} />
          <TutorialScreen image={images.tutorial3} />
          <TutorialScreen image={images.tutorial4} />
        </Swiper>
        <View style={styles.header}>
          <TouchableOpacity>
            <Text style={styles.login}>
              Log In
            </Text>
          </TouchableOpacity>
        </View>
        <View style={styles.footer}>
          <FacebookButton style={buttonStyles.facebookButton} />
          <Button style={buttonStyles.createAccountButton}>
            Create Account
          </Button>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'relative',
    backgroundColor: 'rgba(0, 0, 0, 0)'
  },
  header: {
    position: 'absolute',
    top: '5%',
    right: 0,
    width: '100%',
    height: '20%'
  },
  footer: {
    position: 'absolute',
    bottom: '5%',
    left: 0,
    right: 0,
    width: '100%',
    height: '20%'
  },
  login: {
    fontSize: 18,
    fontFamily: FONT_FAMILY,
    letterSpacing: 0,
    marginRight: 20,
    color: '#fff',
    alignSelf: 'flex-end'
  }
});

const buttonStyles = {
  facebookButton: {},
  createAccountButton: {
    marginTop: 10,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    borderColor: '#fff'
  }
};

export default Tutorial;
