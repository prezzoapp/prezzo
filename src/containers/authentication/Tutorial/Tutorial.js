// @flow
import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TouchableOpacity, StyleSheet, Platform } from 'react-native';
import Swiper from 'react-native-swiper';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { Constants } from 'expo';
import TutorialScreen from './TutorialScreen';
import Button from '../../../components/Button';
import FacebookButton from '../../../components/FacebookButton';
import { FONT_FAMILY_REGULAR } from '../../../services/constants';
import { getUserInfo } from '../../../services/facebook';

const images = {
  tutorial1: require('../../../../assets/images/tutorial/tutorial-1.jpg'),
  tutorial2: require('../../../../assets/images/tutorial/tutorial-2.jpg'),
  tutorial3: require('../../../../assets/images/tutorial/tutorial-3.jpg'),
  tutorial4: require('../../../../assets/images/tutorial/tutorial-4.jpg')
};

class Tutorial extends React.Component {
  static navigationOptions = {
    header: null
  };

  static propTypes = {
    navigate: PropTypes.func.isRequired
  };

  state = {
    isBusy: false
  };

  async onFacebookLogin(facebookId, accessToken) {
    try {
      await this.props.loginWithFacebook(facebookId, accessToken);
      this.navigateToHome();
    } catch (e) {
      if (e && e.code === 404) {
        console.log('user not found');
      }

      try {
        const { email, firstName, lastName, avatarURL } = await getUserInfo(accessToken);

        if (email) {
          this.props.updateEmail(email);
        }

        if (firstName) {
          this.props.updateFirstName(firstName);
        }

        if (lastName) {
          this.props.updateLastName(lastName);
        }

        if (avatarURL) {
          this.props.updateAvatarURL(avatarURL);
        }

        this.props.updateFacebookId(facebookId);
        this.props.updateFacebookToken(accessToken);

        // check if user exists with email
        const user = await this.props.findUser(email);
        this.setState({ isBusy: false });

        if (user) {
          this.navigateToSignupMergeFacebook();
        } else {
          this.navigateToSignup();
        }
      } catch (error) {
        console.log('error getting facebook data', error);
        this.setState({ isBusy: false });
        this.navigateToSignup();
      }
    }
  }

  navigateToSignupMergeFacebook() {
    this.props.navigate({ routeName: 'SignupMergeFacebook' });
  }

  navigateToHome() {
    this.props.navigate({ routeName: 'Customer' });
  }

  navigateToLogin() {
    this.props.navigate({ routeName: 'Login' });
  }

  navigateToSignup() {
    this.props.navigate({ routeName: 'SignupName' });
  }

  navigateToEnableNotifications() {
    this.props.navigate({ routeName: 'EnableNotifications' });
  }

  render() {
    const { isBusy } = this.state;

    return (
      <View style={styles.container}>
        <Swiper
          testID="swiper"
          loop={false}
          dotStyle={styles.dotStyle}
          activeDotStyle={styles.dotStyle}
          paginationStyle={{ bottom: hp('1.72%') }}
          dotColor="rgba(255, 255, 255, 0.5)"
          activeDotColor="#0DD24A"
        >
          <TutorialScreen image={images.tutorial1} />
          <TutorialScreen image={images.tutorial2} />
          <TutorialScreen image={images.tutorial3} />
          <TutorialScreen image={images.tutorial4} />
        </Swiper>
        <View style={styles.header}>
          <TouchableOpacity
            testID="loginButton"
            onPress={() => this.navigateToLogin()}
          >
            <Text style={styles.login}>Log In</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.footer}>
          <FacebookButton
            testID="facebookButton"
            disabled={isBusy}
            style={buttonStyles.facebookButton}
            onStart={() => this.setState({ isBusy: true })}
            onFailure={() => this.setState({ isBusy: false })}
            onCancel={() => this.setState({ isBusy: false })}
            onSuccess={(facebookId, accessToken) => {
              this.onFacebookLogin(facebookId, accessToken);
            }}
          />
          <Button
            testID="signupButton"
            disabled={isBusy}
            textStyle={buttonStyles.textStyle}
            style={buttonStyles.createAccountButton}
            onPress={() => this.navigateToEnableNotifications()}
          >
            Create Account
          </Button>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'relative',
    backgroundColor: 'rgba(0, 0, 0, 0)'
  },
  header: {
    position: 'absolute',
    top: Constants.statusBarHeight + hp('3%') - (Platform.OS === 'ios' ? 20 : 0),
    right: 0,
    width: '100%',
    height: '20%'
  },
  footer: {
    position: 'absolute',
    bottom: hp('4.50%'),
    left: 0,
    right: 0,
    width: '100%'
  },
  login: {
    fontSize: wp('5.33%'),
    fontFamily: FONT_FAMILY_REGULAR,
    letterSpacing: 0,
    marginRight: wp('5.33%'),
    color: '#fff',
    alignSelf: 'flex-end'
  },
  dotStyle: {
    width: wp('1.6%'),
    height: wp('1.6%')
  }
});

const buttonStyles = {
  facebookButton: {
    marginLeft: wp('10.13%'),
    marginRight: wp('10.13%')
  },
  createAccountButton: {
    width: 'auto',
    height: hp('5.29%'),
    marginLeft: wp('10.13%'),
    marginRight: wp('10.13%'),
    justifyContent: 'center',
    marginTop: hp('1.72%'),
    backgroundColor: 'transparent',
    borderColor: '#fff'
  },
  textStyle: {
    paddingTop: 0,
    paddingBottom: 0,
    fontSize: wp('4.8%')
  }
};

export default Tutorial;
