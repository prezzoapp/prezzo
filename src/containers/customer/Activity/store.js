// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';

const mapDispatchToProps = dispatch => {
  return {
    navigate: bindActionCreators(NavigationActions.navigate, dispatch)
  };
};

export default mapDispatchToProps;
