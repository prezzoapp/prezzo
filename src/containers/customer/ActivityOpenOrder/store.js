import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { listOpenOrders } from '../../../modules/userActivity';

export const mapStateToProps = state => {
  const data = state
    .get('userActivity')
    .get('data')
    .toJS();

  const isBusy = state.get('userActivity').get('isBusy');

  return {
    data,
    isBusy
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    listOpenOrders: bindActionCreators(listOpenOrders, dispatch),
    navigate: bindActionCreators(NavigationActions.navigate, dispatch)
  };
};
