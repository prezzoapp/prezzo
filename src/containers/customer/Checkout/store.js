// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';

import { addRemoveItemQuantity, setType } from '../../../modules/restaurant';
import { listCreditCards } from '../../../modules/paymentMethods';
import { userLogout } from '../../../modules/auth';

export const mapStateToProps = state => {
  const data = state.get('restaurant').toJS();
  const type = state.get('restaurant').get('type');
  const paymentType = state.get('restaurant').get('paymentType');
  const newlyAddedCard = state.get('paymentMethods').get('newlyAddedCard');
  const creditCardList = state
    .get('paymentMethods')
    .get('data')
    .toJS();

  return {
    data,
    type,
    paymentType,
    creditCardList,
    newlyAddedCard
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  addRemoveItemQuantity: bindActionCreators(addRemoveItemQuantity, dispatch),
  setType: bindActionCreators(setType, dispatch),
  listCreditCards: bindActionCreators(listCreditCards, dispatch),
  userLogout: bindActionCreators(userLogout, dispatch)
});
