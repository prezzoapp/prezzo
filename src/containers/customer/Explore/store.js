// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { listVendors, toggleFilter, updateDistance, updatePrice, getUserCurrentLocation, callWaiterBtnFunc } from '../../../modules/explore';

export const mapStateToProps = state => {
  const filters = state
    .get('explore')
    .get('filters')
    .toJS();
  const distance = state.get('explore').get('distance');
  const pricing = state.get('explore').get('pricing');
  const isBusy = state.get('explore').get('isBusy');
  const minDistance = state.get('explore').get('minDistance');
  const maxDistance = state.get('explore').get('maxDistance');
  const callWaiterBtnState = state.get('explore').get('callWaiterBtnState')
  const userName = state.get('user').get('account').get('fullName');
  const isUserOpenOrder = state.get('userActivity').get('data').size !== 0 ? true : false

  return {
    filters,
    distance,
    pricing,
    isBusy,
    minDistance,
    maxDistance,
    userName,
    callWaiterBtnState,
    isUserOpenOrder
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  listVendors: bindActionCreators(listVendors, dispatch),
  toggleFilter: bindActionCreators(toggleFilter, dispatch),
  updateDistance: bindActionCreators(updateDistance, dispatch),
  updatePrice: bindActionCreators(updatePrice, dispatch),
  getUserCurrentLocation: bindActionCreators(getUserCurrentLocation, dispatch),
  callWaiterBtnFunc: bindActionCreators(callWaiterBtnFunc, dispatch)
});
