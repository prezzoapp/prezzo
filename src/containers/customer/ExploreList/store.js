// @flow
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { addRestaurantDetail } from '../../../modules/restaurant';

export const mapStateToProps = state => {
  const explore = state.get('explore');
  const restaurants = explore.get('restaurants').toJS();

  return {
    restaurants
  };
};

export const mapDispatchToProps = dispatch => ({
  navigate: bindActionCreators(NavigationActions.navigate, dispatch),
  addRestaurantDetail: bindActionCreators(addRestaurantDetail, dispatch)
});
