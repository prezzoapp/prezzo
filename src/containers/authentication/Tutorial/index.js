// @flow
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { NavigationActions } from 'react-navigation';
import { loginWithFacebook } from '../../../modules/auth';
import { findUser } from '../../../modules/user';
import {
  updateEmail,
  updateFirstName,
  updateLastName,
  updateAvatarURL,
  updateFacebookId,
  updateFacebookToken
} from '../../../modules/Signup';
import Tutorial from './Tutorial';

export default connect(
  null,
  dispatch => {
    return {
      findUser: bindActionCreators(findUser, dispatch),
      updateEmail: bindActionCreators(updateEmail, dispatch),
      updateFirstName: bindActionCreators(updateFirstName, dispatch),
      updateLastName: bindActionCreators(updateLastName, dispatch),
      updateAvatarURL: bindActionCreators(updateAvatarURL, dispatch),
      loginWithFacebook: bindActionCreators(loginWithFacebook, dispatch),
      updateFacebookId: bindActionCreators(updateFacebookId, dispatch),
      updateFacebookToken: bindActionCreators(updateFacebookToken, dispatch),
      navigate: bindActionCreators(NavigationActions.navigate, dispatch)
    };
  }
)(Tutorial);
