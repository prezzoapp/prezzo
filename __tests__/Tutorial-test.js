import 'react-native';
import React from 'react';
import Tutorial from '../src/containers/authentication/Tutorial';
import store from '../src/redux/store';
import { findById } from '../src/services/commonFunctions';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

it('Tourial Screen renders correctly', () => {
  const tutorial = renderer
    .create(
      <Provider store={store}>
        <Tutorial />
      </Provider>
    )
    .toJSON();
  expect(tutorial).toMatchSnapshot();
});

it('it should render a swiper', () => {
  const tutorial = renderer
    .create(
      <Provider store={store}>
        <Tutorial />
      </Provider>
    )
    .toJSON();
  expect(findById(tutorial, 'swiper')).toBeDefined();
});

it('it should render a facebookButton', () => {
  const tutorial = renderer
    .create(
      <Provider store={store}>
        <Tutorial />
      </Provider>
    )
    .toJSON();
  expect(findById(tutorial, 'facebookButton')).toBeDefined();
});

it('it should render a signup button', () => {
  const tutorial = renderer
    .create(
      <Provider store={store}>
        <Tutorial />
      </Provider>
    )
    .toJSON();
  expect(findById(tutorial, 'signupButton')).toBeDefined();
});

it('it should render a login button', () => {
  const tutorial = renderer
    .create(
      <Provider store={store}>
        <Tutorial />
      </Provider>
    )
    .toJSON();
  expect(findById(tutorial, 'loginButton')).toBeDefined();
});
