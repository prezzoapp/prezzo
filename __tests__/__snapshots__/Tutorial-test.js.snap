// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tourial Screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0)",
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      activeDotColor="#0DD24A"
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDirection={true}
      autoplayTimeout={2.5}
      bounces={false}
      contentContainerStyle={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      contentOffset={
        Object {
          "x": 0,
        }
      }
      disableNextButton={false}
      dotColor="rgba(255, 255, 255, 0.5)"
      horizontal={true}
      index={0}
      loadMinimal={false}
      loadMinimalSize={1}
      loop={false}
      onIndexChanged={[Function]}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollsToTop={false}
      showsButtons={false}
      showsHorizontalScrollIndicator={false}
      showsPagination={true}
      showsVerticalScrollIndicator={false}
      style={undefined}
      testID="swiper"
    >
      <View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "bottom": 25,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#0DD24A",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.5)",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.5)",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.5)",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "height": "20%",
        "position": "absolute",
        "right": 0,
        "top": "5%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="loginButton"
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "System",
            "fontSize": 18,
          }
        }
      >
        Log In
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": "5%",
        "height": "20%",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(58,159,249,0.81)",
          "borderColor": "#2074FF",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginLeft": 20,
          "marginRight": 20,
          "opacity": 1,
          "width": "90%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginRight": 15,
            "resizeMode": "contain",
            "width": 25,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "System",
            "fontSize": 20,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Continue With Facebook
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 2,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 10,
          "opacity": 1,
          "width": "90%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "System",
            "fontSize": 20,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Create Account
      </Text>
    </View>
  </View>
</View>
`;
